src/Microsoft.TemplateEngine.Core/Expressions/Shared/XmlStyleConverters.cs

```diff
@@ -0,0 +1,150 @@
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Microsoft.TemplateEngine.Core.Expressions.Shared
+{
+    public static class XmlStyleConverters
+    {
+        public static string XmlDecode(string arg)
+        {
+            List<char> output = new List<char>();
+
+            for (int i = 0; i < arg.Length; ++i)
+            {
+                //Not entity mode
+                if (arg[i] != '&')
+                {
+                    output.Add(arg[i]);
+                    continue;
+                }
+
+                ++i;
+                //Entity mode, decimal or hex
+                if (arg[i] == '#')
+                {
+                    ++i;
+
+                    //Hex entity mode
+                    if (arg[i] == 'x')
+                    {
+                        string hex = arg.Substring(i + 1, 4);
+                        char c = (char)short.Parse(hex.TrimStart('0'), NumberStyles.HexNumber);
+                        output.Add(c);
+                        i += 5; //x, 4 digits, semicolon (consumed by the loop bound)
+                    }
+                    else
+                    {
+                        string dec = arg.Substring(i, 4);
+                        char c = (char)short.Parse(dec.TrimStart('0'), NumberStyles.Integer);
+                        output.Add(c);
+                        i += 4; //4 digits, semicolon (consumed by the loop bound)
+                    }
+                }
+                else
+                {
+                    switch (arg[i])
```

Comment:
Why not use string comparisons instead of these embedded 1-case switches?

Also - either way - do we need bounds checks here, similar to above?